### PR TITLE
Simplify the python base API in a few places

### PR DESCRIPTION
**Pull request by nafur**
This PR simplifies a few methods in the python base API.

### CODE REVIEW

**Review by yoni206 — COMMENTED** [review]
This LGTM but before merging are you able to add some tests for the changes? Unless all possible ways to call these functions are already tested?

**Comment by nafur** [comment]
> This LGTM but before merging are you able to add some tests for the changes? Unless all possible ways to call these functions are already tested?

I added some new tests for mkReal. All other changes either do not change the actual interface.

File: src/api/python/cvc5.pxi [hunk 1]
```diff
@@ -1196,37 +1196,23 @@ cdef class Solver:
...
-            :param val: The value of the term. Can be an integer, float, or
-                        string. It will be formatted as a string before the
-                        term is built.
-            :param den: If not None, the value is ``val``/``den``.
-            :return: A real term with literal value.
+    def mkReal(self, numerator, denominator=None):
+        """Create a real constant from a numerator and an optional denominator.
+        First converts the arguments to a temporary string, either ``"<numerator>"`` or
+        ``"<numerator>/<denominator>"``.
+        This temporary string is forwarded to :cpp:func:`cvc5::Solver::mkReal(const std::string&)`
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:1203]: Can you please format the docstring consistently with the other docstrings?

File: src/api/python/cvc5.pxi [hunk 2]
```diff
@@ -1196,37 +1196,24 @@ cdef class Solver:
...
-    def mkReal(self, val, den=None):
+    def mkReal(self, numerator, denominator=None):
         """
-            Create a real constant.
+            Create a real constant from a numerator and an optional denominator.
+            First converts the arguments to a temporary string, either
+            ``"<numerator>"`` or ``"<numerator>/<denominator>"``. This temporary
+            string is forwarded to
+	    :cpp:func:`cvc5::Solver::mkReal()` and should thus
+	    represent an integer, a decimal number or a fraction.
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:1206]:
```suggestion
            Create a real constant from a numerator and an optional denominator.
            First converts the arguments to a temporary string, either
            ``"<numerator>"`` or ``"<numerator>/<denominator>"``. This temporary
            string is forwarded to :cpp:func:`cvc5::Solver::mkReal()` and should
            thus represent an integer, a decimal number or a fraction.
```

File: src/api/python/cvc5.pxi [hunk 3]
```diff
@@ -1196,37 +1196,24 @@ cdef class Solver:
...
-              ``(W * P + D)/P``, where ``P`` is the appropriate power of 10.
-            - Given a float ``f``, constructs the rational matching ``f``'s
-              string representation. This means that ``mkReal(0.3)`` gives
-              ``3/10`` and not the IEEE-754 approximation of ``3/10``.
-            - Given a string ``"W"`` or an integer, constructs that integer.
-            - Given two strings and/or integers ``N`` and ``D``, constructs
-              ``N/D``.
+            :param numerator: The numerator.
+            :param denominator: The denominator, or ``None``.
+            :return: a real term with literal value
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:1210]:
```suggestion
            :return: A real term with literal value.
```

File: src/api/python/cvc5.pxi [hunk 4]
```diff
@@ -2698,27 +2666,19 @@ cdef class Solver:
...
-            Supports the following variants:
-
-            - ``Term getInterpolant(Term conj)``
-            - ``Term getInterpolant(Term conj, Grammar grammar)``
-
             .. warning:: This method is experimental and may change in future
-                         versions.
-
+                        versions.
+
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:2671]:
```suggestion

```

File: src/api/python/cvc5.pxi [hunk 5]
```diff
@@ -2749,8 +2709,8 @@ cdef class Solver:
         cdef Term result = Term(self)
         result.cterm = self.csolver.getInterpolantNext()
         return result
-
-    def getAbduct(self, Term conj, *args):
+
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:2712]:
```suggestion

```

File: src/api/python/cvc5.pxi [hunk 6]
```diff
@@ -2764,27 +2724,19 @@ cdef class Solver:
...
 
-            Supports the following variants:
-
-            - ``Term getAbduct(Term conj)``
-            - ``Term getAbduct(Term conj, Grammar grammar)``
-
             .. warning:: This method is experimental and may change in future
                          versions.
-
+
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:2729]:
```suggestion

```

File: src/api/python/cvc5.pxi [hunk 7]
```diff
@@ -1196,37 +1196,23 @@ cdef class Solver:
             term.cterm = self.csolver.mkInteger((<int?> val))
         return term
 
-    def mkReal(self, val, den=None):
+    def mkReal(self, numerator, denominator=None):
         """
-            Create a real constant.
+            Create a real constant from a numerator and an optional denominator.
```
Comment (aniemetz) [inline src/api/python/cvc5.pxi:1201]:
```suggestion
            Create a real constant from a numerator and an optional denominator.
            
```

**Review by aniemetz — APPROVED** [review]

**Pull request merged**